samples/cellular/modem_shell/src/location/location_details.c

```diff
@@ -0,0 +1,718 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include <zephyr/shell/shell.h>
+#include <date_time.h>
+#include <cJSON.h>
+#include <math.h>
+#include <getopt.h>
+#include <modem/lte_lc.h>
+#include <net/nrf_cloud.h>
+
+#include <modem/location.h>
+
+#include "mosh_print.h"
+#include "link_api.h"
+#include "link_shell_print.h"
+#include "location_cmd_utils.h"
+#include "location_details.h"
+
+static int json_add_num_cs(cJSON *parent, const char *str, double item)
+{
+	if (!parent || !str) {
+		return -EINVAL;
+	}
+
+	return cJSON_AddNumberToObjectCS(parent, str, item) ? 0 : -ENOMEM;
+}
+
+#define GNSS_PVT_KEY_LAT "lat"
+#define GNSS_PVT_KEY_LON "lng"
+#define GNSS_PVT_KEY_ACC "acc"
+
+#define GNSS_PVT_KEY_ALTITUDE "alt"
+#define GNSS_PVT_KEY_ALTITUDE_ACC "altAcc"
+#define GNSS_PVT_KEY_SPEED "spd"
+#define GNSS_PVT_KEY_SPEED_ACC "spdAcc"
+#define GNSS_PVT_KEY_VER_SPEED "verSpd"
+#define GNSS_PVT_KEY_VER_SPEED_ACC "verSpdAcc"
+#define GNSS_PVT_KEY_HEADING "hdg"
+#define GNSS_PVT_KEY_HEADING_ACC "hdgAcc"
+
+#define GNSS_PVT_KEY_PDOP "pdop"
+#define GNSS_PVT_KEY_HDOP "hdop"
+#define GNSS_PVT_KEY_VDOP "vdop"
+#define GNSS_PVT_KEY_TDOP "tdop"
+#define GNSS_PVT_KEY_FLAGS "flags"
+
+static int location_details_position_encode(
+	const struct location_data *location, cJSON * const root_obj)
+{
+	cJSON *pvt_data_obj = NULL;
+
+	if (!location || !root_obj) {
+		return -EINVAL;
+	}
+
+	pvt_data_obj = cJSON_CreateObject();
+	if (pvt_data_obj == NULL) {
+		mosh_error("No memory to create json obj for position");
+		goto cleanup;
+	}
+
+	if (!cJSON_AddItemToObject(root_obj, "position", pvt_data_obj)) {
```

Comment:
Yep, this is something I didn't change yet because....
Current definitions are what existed in our internal version. Personally, I would probably remove all those definitions and just use string literals where needed. This is simply because I'm not a big fan of definitions for things used just one single time.
I understand this is against most coding guidance so I'll do here whatever the audience instructs 😄

Reply:
`lat`, `lng`, `acc` are used twice.
I'm still going without definitions as that's more readable in my opinion. Let's see if it gets through 😄